### DISCUSSION TITLE
Average speed widget - speed drops to zero during stops, when recording a track

### DISCUSSION THREAD

**Discussion by AndyB69** [osmandapp/OsmAnd#18106] [General]
Hello,

I've got the average speed widget set up in my cycling profile but it's not working as I would want it to.

What I want, is for it to show the average speed over the entire recording track from the start to the current point in time, which could be several hours later.

- I've got "take stops into account" turned OFF (i.e. don't count speeds of speeds less than 1km/hr)
- And I've tried "Time intervals" from 1 min to 60 mins.

Here are my current settings:

<img width="443" alt="image" src="https://github.com/osmandapp/OsmAnd/assets/68859219/2cc7dc82-ee13-47a8-8c31-08bfeec4fb04">

But what seems to be happening, every time I record, is that when I stop the average speed drops, and continues dropping, such that at the end of a longer break (say 10 mins, I'm not sure exactly), the average speed has dropped to zero.

i.e. even though I've got "take stops into account" turned OFF, it does seem to be counting them.

I don't know, of course, but I wonder whether what might be happening is that the GPS inaccuracy, when I'm stopped, causes this type of effect, which then leads OsmAnd to think I'm moving faster than 1km/hr. The screenshot below is a track recorded when the phone is 100% stationary, for example:

<img width="351" alt="image" src="https://github.com/osmandapp/OsmAnd/assets/68859219/0c14ed6b-1d4c-49fa-ab86-7971981d5122">

I've NOT tried the track recording options yet - to see if they make a difference - wanted to check with you first: 

- Logging "Minimum displacement" setting (currently "Not Selected") - would that affect the Average Speed widget though?

- There is also a Logging "Minimum Speed" setting (currently "Not Selected") - but would this affect the Average speed widget which seems like it's hard-coded to 1kph?

- And how about the Logging "Auto split recordings after gap" (I have this ON) - is it that for longer breaks, a new segment is being created and the average speed widget works per segment, rather than for a recording as a whole? I'm not actually sure what a segment is or how that is shown in a recording...

Many thanks, 
Andrew

**Comment by vshcherb** [comment]
Logging is totally unrelated from widget functionality. All individual measurements according to documentation, so if you stop average will anyway will decrease to 0 over time or will be replace with No data value

**Reply by AndyB69** [reply]
Yeah, thanks, I just figured that - what I need is a trip-recording average speed widget, not this one. 

Regards, 
Andrew
